gio/emit.go

```diff
@@ -14,11 +14,15 @@ func Emit(anyObject ...interface{}) error {
 // TsEmit encode and write a row of data to os.Stdout
 // with ts in milliseconds epoch time
 func TsEmit(ts int64, anyObject ...interface{}) error {
-	stat.Stats[0].OutputCounter++
```

Comment:
How about using atomic.Add functions?

Similar for other changes. It'll be nice if this is wrapped in a function to avoid repeated Lock/Unlock.

Reply:
I don't think it's safe enough, since you're also doing other more complex modifications, such as in `runner.go`:

```go
	stat.v.FlowHashCode = uint32(runner.Option.HashCode)
	stat.v.Stats = []*pb.InstructionStat{
		{
			StepId: int32(runner.Option.StepId),
			TaskId: int32(runner.Option.TaskId),
		},
	}
```

Reply:
the stats would not change after setting the Stats.

I am more concerned about the lock/unlock cost for this counter.